src/main/java/net/mcreator/ui/blockly/BlocklyJavascriptBridge.java

```diff
@@ -139,6 +140,16 @@ public class BlocklyJavascriptBridge {
 			case "spawnableEntity" -> openDataListEntrySelector(
 					w -> ElementUtil.loadAllSpawnableEntities(w).stream().filter(e -> e.isSupportedInWorkspace(w))
 							.toList(), "entity");
+			case "customEntity" -> openDataListEntrySelector(ElementUtil::loadCustomEntities, "entity");
+			case "entitydata_logic" -> openStringEntrySelector(
+					w -> ElementUtil.loadEntityDataListFromCustomEntity(w, customEntryProviders,
+							PropertyData.LogicType.class).toArray(String[]::new), "entity");
+			case "entitydata_number" -> openStringEntrySelector(
+					w -> ElementUtil.loadEntityDataListFromCustomEntity(w, customEntryProviders,
+							PropertyData.NumberType.class).toArray(String[]::new), "entity");

```

Comment:
This probably needs to be changed to "PropertyData.IntegerType" as I got confused by the name entitydata_number and set the wrong type (or right based on the name entitydata_number)